src/discord_integration.py

```diff
@@ -70,36 +71,45 @@ def get_messages(channel_id: int, limit: int = 100) -> list:
         f"{api_base}/channels/{channel_id}/messages?limit={limit}", headers=headers
     )
 
-    jsonn = json.loads(r.text)
     new_list = []
 
     if r.status_code != 200:
         new_list.append(
             {
+                "timestamp": datetime.now(),
                 "username": "System",
                 "content": "Error. This may be a forum channel, or you're not allowed to view the content.",
                 "id": 0,
             }
         )
         return new_list
 
-    for value in jsonn:
+    for value in r.json():
```

Comment:
Maybe we should rename value to "message" for clarity. Just looking at my terrible variable names :)